src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirOperationDataStore.cs

```diff
@@ -71,21 +72,93 @@ public async Task<ExportJobOutcome> CreateExportJobAsync(ExportJobRecord jobReco
+
+                using (SqlDataReader sqlDataReader = await sqlCommand.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
+                {
+                    if (!sqlDataReader.Read())
+                    {
+                        throw new JobNotFoundException(string.Format(Core.Resources.JobNotFound, id));
+                    }
+
+                    (string rawJobRecord, byte[] rowVersion) = sqlDataReader.ReadRow(V1.ExportJob.RawJobRecord, V1.ExportJob.JobVersion);
+
+                    return CreateExportJobOutcome(rawJobRecord, rowVersion);
+                }
+            }
         }
 
-        public Task<ExportJobOutcome> GetExportJobByHashAsync(string hash, CancellationToken cancellationToken)
+        public async Task<ExportJobOutcome> GetExportJobByHashAsync(string hash, CancellationToken cancellationToken)
         {
-            // TODO: Implement this method (returning null for now to allow the create method to run).
-            return Task.FromResult<ExportJobOutcome>(null);
+            EnsureArg.IsNotNullOrWhiteSpace(hash, nameof(hash));
+
+            using (SqlConnectionWrapper sqlConnectionWrapper = _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapper(true))
+            using (SqlCommand sqlCommand = sqlConnectionWrapper.CreateSqlCommand())
+            {
+                V1.GetExportJobByHash.PopulateCommand(sqlCommand, hash);
+
+                using (SqlDataReader sqlDataReader = await sqlCommand.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
+                {
+                    if (!sqlDataReader.Read())
+                    {
+                        return null;
+                    }
+
+                    (string rawJobRecord, byte[] rowVersion) = sqlDataReader.ReadRow(V1.ExportJob.RawJobRecord, V1.ExportJob.JobVersion);
+
+                    return CreateExportJobOutcome(rawJobRecord, rowVersion);
+                }
+            }
         }
 
-        public Task<ExportJobOutcome> UpdateExportJobAsync(ExportJobRecord jobRecord, WeakETag eTag, CancellationToken cancellationToken)
+        public async Task<ExportJobOutcome> UpdateExportJobAsync(ExportJobRecord jobRecord, WeakETag eTag, CancellationToken cancellationToken)
         {
-            // TODO: Implement this method.
-            return Task.FromResult(new ExportJobOutcome(jobRecord, eTag));
+            EnsureArg.IsNotNull(jobRecord, nameof(jobRecord));
+
+            byte[] rowVersionAsBytes = GetRowVersionAsBytes(eTag);
+
+            using (SqlConnectionWrapper sqlConnectionWrapper = _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapper(true))
+            using (SqlCommand sqlCommand = sqlConnectionWrapper.CreateSqlCommand())
+            {
+                V1.UpdateExportJob.PopulateCommand(
+                    sqlCommand,
+                    jobRecord.Id,
+                    jobRecord.Status.ToString(),
+                    jobRecord.QueuedTime,
```

Comment:
`QueuedTime` will always remain the same for an export job. I don't think we need to pass it in for the update call.

Reply:
Good point. I thought about this and originally ended up leaving it in because I used it to prioritize older queued jobs when acquiring export jobs in the `dbo.AcquireExportJobs` procedure.

Looking at the Cosmos implementation, I noticed that the queued jobs are timestamped (the same way I timestamp running jobs), and this is used when deciding which jobs to pick up. I took out `QueuedDatetime` from the `dbo.ExportJob` table and updated `HeartbeatDateTime` in `dbo.CreateExportJob` in bd0d061. This simplifies the SQL side implementation and aligns its behaviour with the Cosmos side. Thank you!